.github/workflows/fe-build-and-test.yaml

```diff
@@ -0,0 +1,36 @@
+name: FE Test and Build
+
+on:
+  push:
+    branches:
+      - '!master'
+    paths:
+      - thirdeye-ui/**
+  workflow_dispatch:
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Pull Repository
+        uses: actions/checkout@v4
+        with:
+          sparse-checkout: |
+            thirdeye-ui
+            pom.xml
```

Comment:
curious on why you need pom.xml and .mvn ? 
I remember some time ago we made the java build tools and the frontend tools totally separated.

Reply:
I was referencing the publish action and in that its used for distribution step. And yes we don't need it in this action. Will remove it.